inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -654,12 +653,26 @@ void RemovePermutationsNHWCToNCHWPass::run() {
         auto next = getInputTo(l->outData.front()).begin()->second;
         auto prev = CNNNetPrevLayer(l);
 
-        if (!LayerInfo(next).isPermute() || !LayerInfo(prev).isPermute()) {
+        // The next layer must be NCHW to NHWC permute
+        if (!LayerInfo(next).isPermute() || next->GetParamAsInts("order") != std::vector<int>({0, 2, 3, 1})) {
             continue;
         }
 
-        if (getPassManager()->getPolicy().NHWCToNCHWPolicy == Policy::NHWCToNCHW::REMOVE_ALL) {
-            permutationsToRemove.push_back(prev);
+        // The previous layer must be NHWC to NCHW permute or have 1D data
+        if (LayerInfo(prev).isPermute()) {
+            if (prev->GetParamAsInts("order") != std::vector<int>({0, 3, 1, 2})) continue;
```

Comment:
please check layout instead of order of channels or use utility function (similar to getFromIRDimsOrderNCHW?) to return order.